### ISSUE TITLE
applyConfiguration is sometimes time-consuming

### ISSUE THREAD

**Issue by jmirabel** [humanoid-path-planner/gepetto-viewer-corba#50]
After some testing, it seems the problem comes from the unnecessary lock. This means some thread acquires the lock and do not release it on time.

@andreadelprete, were you able to repeat this behaviour with `gepetto-gui` ?

**Comment by andreadelprete** [comment]
Yes, actually with gepetto-gui the times are much worse than with gepetto-viewer. 
These are the computation time I got using gepetto-gui:
```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
       94    0.000    0.000    0.009    0.000 graphical_interface_idl.py:221(refresh)
        1    0.000    0.000    0.027    0.027 graphical_interface_idl.py:290(addSphere)
      993    0.004    0.000   13.270    0.013 graphical_interface_idl.py:344(applyConfiguration)
        1    0.000    0.000    0.033    0.033 graphical_interface_idl.py:371(setLightingMode)
       31    0.002    0.000    0.002    0.000 robot_wrapper.py:120(updateGeometryPlacements)
      961    0.003    0.000    0.003    0.000 robot_wrapper.py:134(viewerNodeNames)
       31    0.045    0.001   13.179    0.425 robot_wrapper.py:194(display)
       31    0.001    0.000    0.001    0.000 robot_wrapper.py:69(com)
        1    0.002    0.002   13.887   13.887 viewer_utils.py:101(play)
        1    0.000    0.000    0.061    0.061 viewer_utils.py:173(addSphere)
        1    0.000    0.000    0.000    0.000 viewer_utils.py:25(xyzRpyToViewerConfig)
       31    0.000    0.000    0.273    0.009 viewer_utils.py:253(updateObjectConfig)
       31    0.001    0.000   13.180    0.425 viewer_utils.py:74(updateRobotConfig)
     1089   13.335    0.012   13.335    0.012 {_omnipy.invoke}
```
You can see that ```applyConfiguration``` takes on average 13 ms.

**Comment by andreadelprete** [comment]
When using gepetto-viewer I got these times instead:
```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
       94    0.000    0.000    0.667    0.007 graphical_interface_idl.py:221(refresh)
        1    0.000    0.000    0.000    0.000 graphical_interface_idl.py:290(addSphere)
      993    0.003    0.000    1.091    0.001 graphical_interface_idl.py:344(applyConfiguration)
        1    0.000    0.000    0.000    0.000 graphical_interface_idl.py:371(setLightingMode)
       31    0.002    0.000    0.002    0.000 robot_wrapper.py:120(updateGeometryPlacements)
      961    0.002    0.000    0.002    0.000 robot_wrapper.py:134(viewerNodeNames)
       31    0.036    0.001    1.284    0.041 robot_wrapper.py:194(display)
       31    0.001    0.000    0.001    0.000 robot_wrapper.py:69(com)
        1    0.002    0.002    2.573    2.573 viewer_utils.py:101(play)
        1    0.000    0.000    0.001    0.001 viewer_utils.py:173(addSphere)
        1    0.000    0.000    0.000    0.000 viewer_utils.py:25(xyzRpyToViewerConfig)
       31    0.000    0.000    0.013    0.000 viewer_utils.py:253(updateObjectConfig)
       31    0.001    0.000    1.284    0.041 viewer_utils.py:74(updateRobotConfig)
     1089    1.755    0.002    1.755    0.002 {_omnipy.invoke}
```
In this case ```applyConfiguration``` takes only 1 ms on average.

**Comment by jmirabel** [comment]
b4ee39e should fix the issue. I'll merge it when I get some time.

**Comment by jmirabel** [comment]
I think this should be fixed by #51. Note that you must update gepetto-viewer as well.

**Comment by jmirabel** [comment]
If your issue was solved, could you close this issue ?
